service/src/java/org/apache/hive/service/cli/CLIService.java

```diff
@@ -457,9 +469,34 @@ public OperationStatus getOperationStatus(OperationHandle opHandle)
     }
     OperationStatus opStatus = operation.getStatus();
     LOG.debug(opHandle + ": getOperationStatus()");
+    opStatus.setJobProgressUpdate(progressUpdateLog(getProgressUpdate, operation));
     return opStatus;
   }
 
+  private JobProgressUpdate progressUpdateLog(boolean isProgressLogRequested, Operation operation) {
+    if (isProgressLogRequested && canProvideProgressLog()) {
+        if (OperationType.EXECUTE_STATEMENT.equals(operation.getType())) {
+          SessionState sessionState = operation.getParentSession().getSessionState();
+          try {
+            while (sessionState.progressMonitor() == null && !operation.isFinished()) {
+              Thread.sleep(100L); // sleep for 100 ms
```

Comment:
I can not sleep and that will lead to spin lock kind of usage scenario, with cpu being used extensively, this should almost never happen now that we only call this after waiting for the operation status is available, but will be good to have it here since we dont get guarantee that progress monitor is updated.